text/3451-additional-float-types.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: `additional-float-types`
+- Start Date: 2023-6-28
+- RFC PR: [rust-lang/rfcs#3451](https://github.com/rust-lang/rfcs/pull/3451)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes new floating point types `f16` and `f128` into core language and standard library. Also this RFC introduces `f80`, `doubledouble`, `bf16` into `core::arch` for inter-op with existing native code.
```

Comment:
```suggestion
This RFC proposes new floating point types `f16` and `f128` into core language and standard
library. Also, this RFC introduces `f80`, `doubledouble`, and `bf16` into `core::arch` for
target-specific support, and `core::ffi::c_longdouble` for FFI interop.
```

Reply:
Suggest use these symbols, all start with `f` prefix that consistence
- `f128`: C `_Float128`, LLVM `fp128`, GCC `__float128`
- `f16`: C `_Float16`, LLVM `half`, GCC `__fp16`
- `f16b`: C++ `std::bfloat16_t`, LLVM `bfloat`, GCC `__bf16`
- `f80x`: LLVM `x86_fp80`, GCC `__float80`
- `f128x`: LLVM `ppc_fp128`, GCC `__ibm128`

Reply:
+1 for `f16b` in favor over `bf16` for consistency, I liked that about @joshtriplett's original proposal.

I don't think we should introduce something like `f128x` - `doubledouble` or something like the GCC or LLVM types would be better IMO. Reason being, it's kind of ambiguous and specific to one architecture - PowerPC is [even moving away from it](https://fedoraproject.org/wiki/Changes/PPC64LE_Float128_Transition). Better to give it an unambigous name since it will be used relatively rarely.

Reply:
i think we should use `bf16` rather than `f16b` since that is widely recognized whereas `f16b` isn't, and `f64_f64` instead of `f128x` since it really is 2 `f64` values and could be easily emulated on any other architecture (do not use `f64x2` since that's already used by `Simd<f64, 2>`). also `f<N>x` names are more or less defined by IEEE 754 to be wider than `N` bits, so e.g. `f64x` would be approximately any type wider than `f64` but less than `f128` such as `f80`, `f16x` could be the `f24` type used by some GPUs for depth buffers. so logically `f80x` would need to be more than 80 bits and `f128x` would need to be more than 128 bits.

Reply:
> +1 for `f16b` in favor over `bf16` for consistency, I liked that about @joshtriplett's original proposal.
> 
> I don't think we should introduce something like `f128x` - `doubledouble` or something like the GCC or LLVM types would be better IMO. Reason being, it's kind of ambiguous and specific to one architecture - PowerPC is [even moving away from it](https://fedoraproject.org/wiki/Changes/PPC64LE_Float128_Transition). Better to give it an unambigous name since it will be used relatively rarely.

`f64f64` is OK, no need the underscore looks like `doubledouble`, `f80e` instead `f80x` if `f80x` is not acceptable

Reply:
What about `f64x2` to indicate it's two f64 glued together?

Reply:
(do not use f64x2 since that's already used by Simd<f64, 2>)